File: src/test/java/climbstairs/SolutionTest.java
package climbstairs;

import org.assertj.core.api.Assertions;
import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.CsvSource;

class SolutionTest {

    private final Solution solution = new Solution();

    @ParameterizedTest
    @CsvSource(value = {"0:1", "1:1", "2:2", "3:3"}, delimiter = ':')
    public void shouldPass(String input, String output) {
        int noOfWays = solution.climbStairs(Integer.parseInt(input));

        Assertions.assertThat(noOfWays).isEqualTo(Integer.valueOf(output));
    }

}

File: src/test/kotlin/dev/sanket/isomorphicstrings/SolutionTest.kt
package dev.sanket.isomorphicstrings

import assertk.assertThat
import assertk.assertions.isEqualTo
import dev.sanket.dev.sanket.isomorphicstrings.Solution
import org.junit.jupiter.params.ParameterizedTest
import org.junit.jupiter.params.provider.CsvSource

class SolutionTest {
    private val solution = Solution()

    @ParameterizedTest
    @CsvSource(value = ["egg:add:true", "egg:egg:true", "egg:hello:false", "badc:baba:false"], delimiter = ':')
    fun shouldPass(
        s: String,
        t: String,
        isIsMorphic: Boolean,
    ) {
        assertThat(solution.isIsomorphic(s, t)).isEqualTo(isIsMorphic)
    }
}
